### PR TITLE
EN-4198 : add new metric in termui for final highest block nonce

### PR DESCRIPTION
**Pull request by miiu96**
Added a new metric in termui.
Metric contains nonce of final highest block in current shard.

### CODE REVIEW

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go [hunk 1]
```diff
@@ -239,6 +239,12 @@ func (wr *WidgetsRender) prepareBlockInfo() {
 	crossCheckBlockHeight := wr.presenter.GetCrossCheckBlockHeight()
 	rows[4] = []string{fmt.Sprintf("Cross check block height: %s", crossCheckBlockHeight)}
 
+	shardId := wr.presenter.GetShardId()
+	if shardId != uint64(sharding.MetachainShardId) {
+		highestBlockInShard := wr.presenter.GetHighestFinalBlockInShard()
+		rows[4][0] += fmt.Sprintf(" ,highest block nonce in shard: %d", highestBlockInShard)
```
Comment (SebastianMarian) [inline statusHandler/view/termuic/termuiRenders/widgetsRender.go]: ,highest final block nonce in shard

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go [hunk 2]
```diff
@@ -239,6 +239,12 @@ func (wr *WidgetsRender) prepareBlockInfo() {
 	crossCheckBlockHeight := wr.presenter.GetCrossCheckBlockHeight()
 	rows[4] = []string{fmt.Sprintf("Cross check block height: %s", crossCheckBlockHeight)}
 
+	shardId := wr.presenter.GetShardId()
+	if shardId != uint64(sharding.MetachainShardId) {
+		highestBlockInShard := wr.presenter.GetHighestFinalBlockInShard()
```
Comment (SebastianMarian) [inline statusHandler/view/termuic/termuiRenders/widgetsRender.go]: highestFinalBlockInShard

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go (hunk 1 above)
Comment (miiu96) [reply statusHandler/view/termuic/termuiRenders/widgetsRender.go]: done.

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go (hunk 2 above)
Comment (miiu96) [reply statusHandler/view/termuic/termuiRenders/widgetsRender.go]: done.

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go (hunk 1 above)
Comment (bogdan-rosianu) [inline statusHandler/view/termuic/termuiRenders/widgetsRender.go]: add a space after the comma and also set the default value for previous metric

File: core/constants.go [hunk 3]
```diff
@@ -152,3 +152,6 @@ const MetricNumShardHeadersProcessed = "erd_num_shard_headers_processed"
 
 // MetricNumTimesInForkChoice is the metric that counts how many time a node was in fork choice
 const MetricNumTimesInForkChoice = "erd_fork_choice_count"
+
+//MetricHighestFinalBlockInShard is the metric that store highest nonce block notarized by meta for current shard
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/constants.go]: done.

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go (hunk 1 above)
Comment (miiu96) [reply statusHandler/view/termuic/termuiRenders/widgetsRender.go]: done.

**Pull request merged**